hudi-common/src/main/java/org/apache/hudi/common/table/log/InstantRange.java

```diff
@@ -57,6 +59,15 @@ public String getEndInstant() {
 
   public abstract boolean isInRange(String instant);
 
+  @Override
+  public String toString() {
+    return "InstantRange{"
```

Comment:
Use `this.getClass().getName()` here instead of adding `rangeType`?  That should be enough to differentiate different range types.

Reply:
Done.

Reply:
The start or end range may be null.

Reply:
> The start or end range may be null.

It won't.

https://github.com/apache/hudi/blob/98d956fd8456dac8f04dea28a3721527aef705ec/hudi-common/src/main/java/org/apache/hudi/common/table/log/InstantRange.java#L35-L41

Constructors must assign values to start and end range, unless InstantRange itself is null , It will print like this: instantRange=Optional.empty

`[split_reader -> Sink: Collect table sink (1/1)#0] INFO org.apache.hudi.source.StreamReadOperator - Processing input split : MergeOnReadInputSplit{splitNum=1, basePath=Option{val=hdfs://nameservice/hive/warehouse/hudi_test.db/hudi_cow_1024_a/2023-10-24/20/2/eb710199-e526-4e93-ae56-d46adff39eb5-0_0-1-0_20231024202605303.parquet}, logPaths=Option{val=[]}, latestCommit='20231025115243168', tablePath='hdfs://nameservice/hive/warehouse/hudi_test.db/hudi_cow_1024_a', maxCompactionMemoryInBytes=104857600, mergeType='payload_combine', instantRange=Optional.empty}`

Reply:
@danny0405 is right, either `startInstant` or `endInstant` can be null.

Reply:
e.g., `CloseCloseRangeNullableBoundary`.  There's logic checking whether `startInstant` or `endInstant` is null when checking `isInRange`.